napari/components/dims.py

```diff
@@ -295,13 +348,17 @@ def set_current_step(
                 raise ValueError(
                     trans._("axis and value sequences must have equal length")
                 )
-            if value != full_current_step:
-                # (computed) nsteps property outside of the loop for efficiency
-                nsteps = self.nsteps
-                for ax, val in zip(axis, value):
-                    ax = assert_axis_in_bounds(int(ax), self.ndim)
-                    step = round(min(max(val, 0), nsteps[ax] - 1))
+            # (computed) nsteps property outside of the loop for efficiency
+            nsteps = self.nsteps
+            needs_update = False
+            for ax, val in zip(axis, value):
+                ax = assert_axis_in_bounds(int(ax), self.ndim)
+                step = round(min(max(val, 0), nsteps[ax] - 1))
+                if full_current_step[ax] != step:
                     full_current_step[ax] = step
+                    needs_update = True
+
+            if value != full_current_step or needs_update:
```

Comment:
I think we can skip `needs_update` by changing the condition at the end a little.

```suggestion
            for ax, val in zip(axis, value):
                ax = assert_axis_in_bounds(int(ax), self.ndim)
                step = round(min(max(val, 0), nsteps[ax] - 1))
                if full_current_step[ax] != step:
                    full_current_step[ax] = step

            if np.array_equal(self.current_step, full_current_step):
```

Reply:
I adjusted the conditional statement